### PR TITLE
Extends Target.Dependency to be StringLiteralConvertible

### PR DESCRIPTION
**Pull request by jemmons**
This helps make the manifest file a little less noisy. See: SR-154

### CODE REVIEW

**Comment by kostiakoval** [comment]
:+1: Great. Maybe it would be useful to add an `FunctionalBuildTests` for that functionality

**Comment by jemmons** [comment]
Test case (and associated fixture) added to `FunctionalBuildTests`.

**Comment by jemmons** [comment]
Rebased onto master to resolve conflicts.

File: Sources/PackageDescription/Target.swift [hunk 1]
```diff
@@ -53,6 +53,22 @@ extension Target: TOMLConvertible {
     }
 }
 
+// MARK: StringLiteralConvertible
+
+extension Target.Dependency : StringLiteralConvertible{
```
Comment (kostiakoval) [inline Sources/PackageDescription/Target.swift]: Add space between type and opening `}` bracket. Also use this style in other places.
Like this - `tringLiteralConvertible {`
Let's keep one common style. :)

**Comment by jemmons** [comment]
Fixed. Is there a style guide we should adhere to? Something a little more syntactical than the [API Design Guidelines](https://swift.org/documentation/api-design-guidelines.html)? I can get pretty whitespace-blind without Xcode to help me. :worried:

**Comment by mxcl** [comment]
There is no style guide. But we mirror that of Apple’s Swift documentation.

**Comment by mxcl** [comment]
Can we squash the style adjustments into the main commit.

**Comment by jemmons** [comment]
Squashed!

**Pull request merged**